### PR TITLE
Implement API for claims repository (GDEV-1429)

### PR DESCRIPTION
**Pull request by Cito**

### CODE REVIEW

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Just a few minor things.

Currently, the claims repository and user registry have no dedicated core module but the business logic is mostly contained within the fastapi routes. That should probably be changed in the later refactoring. I will leave a note.

File: auth_service/user_management/user_registry/__init__.py [hunk 1]
```diff
@@ -13,8 +13,9 @@
...
 
-"""Utils for Fixture handling"""
+"""User Registry
 
-from pathlib import Path
+The GHGA users registry holds all of the user related data.
+"""
 
-BASE_DIR = Path(__file__).parent.resolve()
+__version__ = "0.1.0"
```
[the remark this answers is not in the document]
Comment (Cito) [reply auth_service/user_management/user_registry/__init__.py:21]: Removed these individual version numbers for now. The Auth Adapter can still have its own version.

File: auth_service/user_management/claims_repository/router.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    """Revoke an existing user claim"""
+    if await user_exists(user_id, user_dao):
+        try:
+            claim = await claim_dao.get_by_id(claim_id)
+            if claim.user_id != user_id:
+                raise ResourceNotFoundError
+        except ResourceNotFoundError:
+            claim = None
+    else:
+        claim = None
```
[the remark this answers is not in the document]
Comment (Cito) [reply auth_service/user_management/claims_repository/router.py]: Changed in 0cec9e7.

File: auth_service/user_management/claims_repository/router.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            if claim.user_id != user_id:
+                raise ResourceNotFoundError
+            await claim_dao.delete(id_=claim_id)
+        except ResourceNotFoundError:
+            claim = None
+    else:
+        claim = None
+
+    if not claim:
+        raise HTTPException(status_code=404, detail="The user claim was not found.")
```
[the remark this answers is not in the document]
Comment (Cito) [reply auth_service/user_management/claims_repository/router.py]: Changed in 0cec9e7.

File: auth_service/user_management/claims_repository/deps.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+) -> ClaimDaoFactory:
+    """Get claim DAO factory."""
+    return ClaimDaoFactory(config=config, dao_factory=dao_factory)
+
+
+async def get_claim_dao(
+    dao_factory: ClaimDaoFactory = Depends(get_claim_dao_factory),
+) -> ClaimDao:
+    """Get claim data access object."""
+    return await dao_factory.get_claim_dao()
```
[the remark this answers is not in the document]
Comment (Cito) [reply auth_service/user_management/claims_repository/deps.py:51]: Sollte zusammen mit dem Auslagern in ein Core-Modul gemacht werden.

**Comment by Cito** [comment]
Thanks for the review, all good points. Readey for review again.

File: auth_service/user_management/user_registry/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply auth_service/user_management/user_registry/__init__.py]: The problem is that our production deployment will likely be triggered with a new release of the repository and there we can only have one version. However, we can figure that out in a different PR.

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**